File: UtilitiesVB/To Depricate/CSVDictUtilities.vb
Imports System.IO
Imports Microsoft.VisualBasic.FileIO

Public Module CSVDictUtilities

    Public Function LoadDictCSV(stagingPath, filename) As Dictionary(Of String, String)
        Dim filepath As String = Path.Combine(stagingPath, filename)
        Dim dictString As New Dictionary(Of String, String)

        Try
            Using MyReader As New TextFieldParser(filepath)
                MyReader.TextFieldType = FileIO.FieldType.Delimited
                MyReader.SetDelimiters(",")

                Dim currentRow As String()
                While Not MyReader.EndOfData
                    Try
                        currentRow = MyReader.ReadFields()
                        Dim key As Object = currentRow(0)
                        Dim value As Object = currentRow(1)
                        dictString.Add(key, value)
                    Catch ex As MalformedLineException
                        Dim unused2 = MsgBox("Line " & ex.Message &
                    "is not valid and will be skipped.")
                    End Try
                End While
            End Using
        Catch e As FileNotFoundException
            Dim unused1 = MsgBox("File not found error -> " & filepath)
        Catch e As FieldAccessException
            Dim unused = MsgBox("File is in use -> " & filepath)
        End Try

        Return dictString

    End Function

    Public Sub WriteDictCSV(dict_str As Dictionary(Of String, String),
                            staging_path As String,
                            filename As String)
        Dim filepath As String = Path.Combine(staging_path, filename)
        Dim csv As String = String.Join(
            Environment.NewLine,
            dict_str.[Select](Function(d) $"{d.Key};{d.Value};"))
        System.IO.File.WriteAllText(filepath, csv)
    End Sub

End Module

File: UtilitiesVB/To Depricate/CSVDictUtilities.cs
using System;
using System.Collections.Generic;
using System.Data;
using System.IO;
using System.Linq;
using Microsoft.VisualBasic;
using Microsoft.VisualBasic.CompilerServices;
using Microsoft.VisualBasic.FileIO;

namespace UtilitiesVB
{

    public static class CSVDictUtilities
    {

        public static Dictionary<string, string> LoadDictCSV(object stagingPath, object filename)
        {
            string filepath = Path.Combine(Conversions.ToString(stagingPath), Conversions.ToString(filename));
            var dictString = new Dictionary<string, string>();

            try
            {
                using (var MyReader = new TextFieldParser(filepath))
                {
                    MyReader.TextFieldType = FieldType.Delimited;
                    MyReader.SetDelimiters(",");

                    string[] currentRow;
                    while (!MyReader.EndOfData)
                    {
                        try
                        {
                            currentRow = MyReader.ReadFields();
                            object key = currentRow[0];
                            object value = currentRow[1];
                            dictString.Add(Conversions.ToString(key), Conversions.ToString(value));
                        }
                        catch (MalformedLineException ex)
                        {
                            var unused2 = Interaction.MsgBox("Line " + ex.Message + "is not valid and will be skipped.");
                        }
                    }
                }
            }
            catch (FileNotFoundException e)
            {
                var unused1 = Interaction.MsgBox("File not found error -> " + filepath);
            }
            catch (FieldAccessException e)
            {
                var unused = Interaction.MsgBox("File is in use -> " + filepath);
            }

            return dictString;

        }

        public static void WriteDictCSV(Dictionary<string, string> dict_str, string staging_path, string filename)
        {
            string filepath = Path.Combine(staging_path, filename);
            string csv = string.Join(Environment.NewLine, dict_str.Select(d => $"{d.Key};{d.Value};"));
            File.WriteAllText(filepath, csv);
        }

    }
}